model/src/main/kotlin/Defect.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model
+
+import com.fasterxml.jackson.annotation.JsonInclude
+
+import java.net.URI
+import java.time.Instant
+
+/**
+ * A data class defining a model for software defects.
+ *
+ * Instances of this class are created by advisor implementations that retrieve information about known defects in
+ * packages.
+ */
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+data class Defect(
+    /**
+     * The (external) ID of this defect. This is a string used by a concrete issue tracker system to reference this
+     * defect, such as a bug ID or ticket number.
+     */
+    val id: String,
+
+    /**
+     * The URL pointing to the source of this defect. This is typically a reference into the issue tracker system that
+     * contains this defect.
+     */
+    val url: URI,
+
+    /**
+     * A title for this defect. This is a short summary describing the problem at hand.
+     */
+    val title: String,
+
+    /**
+     * A state of the associated defect. The concrete meaning of this string depends on the source from where it was
+     * obtained, as different issue tracker systems use their specific terminology. Possible values could be *OPEN*,
+     * *IN PROGRESS*, *BLOCKED*, etc.
+     */
+    val state: String = UNKNOWN_STATE,
+
+    /**
+     * The severity assigned to the defect. The meaning of this string depends on the source system.
+     */
+    val severity: String = UNKNOWN_SEVERITY,
+
+    /**
+     * An optional description of this defect. It can contain more detailed information about the defect and its
+     * impact. The field may be undefined if the [url] of this defect already points to a website with all this
+     * information.
+     */
+    val description: String? = null,
+
+    /**
+     * The creation date of this defect if available.
+     */
+    val createdAt: Instant? = null,
+
+    /**
+     * Contains a date of last modification of this defect if available. This information can be useful for instance
```

Comment:
to me it's unclear what is a "modification of this defect" exactly?

Reply:
Reworded the comment.